ui-components/src/helpers/filters.ts

```diff
@@ -53,6 +54,7 @@ export interface ListingFilterState {
   [FrontendListingFilterStateKeys.minRent]?: string | number
   [FrontendListingFilterStateKeys.maxRent]?: string | number
   [FrontendListingFilterStateKeys.seniorHousing]?: string | boolean
+  [FrontendListingFilterStateKeys.specialNeedsHousing]?: string | boolean
```

Comment:
Can this just be a boolean? There was some issues with the includeNulls when it was both a string and a boolean.

Reply:
I'm not sure why this has type string. Perhaps Austin can address this in a follow up PR, since he chose this set-up for the front end filters and has more context on that?

Reply:
I don't have strong opinions here, we can just set the type to `boolean` if that's preferred, but I was trying to keep the type system from making promises it couldn't keep:
- When we parse the filters from the url they're read in as strings, and never parsed into their appropriate types. But the type system trusts the parser and assumes it gave it a boolean. So then we have a variable that the type system thinks is a bool, but is actually a string. This will cause subtle bugs if you try to do things like `filterValue == false` when the filterValue is the string `"false"`.
- When the filters are set from the form, they're set with their proper types.
- So rather than fix that by adding logic to properly decode every filter type, I just set the types to `... | string` so we'd know what we actually have.

This won't have the same issue as `includeNulls`. Here, `false` and `unset` are different values, so we keep them both around. With `includeNulls`, `false` and `unset` are the same, so we tried to be clever and unset it if it was false, which seemed to confuse the form hook for some reason.

Reply:
TLDR I think we should leave it the way @elenm has it, and it works for the seniorHousing filter already, but if @willrlin has very strong opinions the other way then we can change it to just a boolean type.

Reply:
I'll also mail a PR that adds a comment to that code explaining why the types are that way.

Reply:
Thanks! Resolving for this PR.